File: autojs/src/main/java/com/stardust/autojs/runtime/accessibility/AccessibilityConfig.java
package com.stardust.autojs.runtime.accessibility;

import com.stardust.util.DeveloperUtils;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Stardust on 2017/4/29.
 */
public class AccessibilityConfig {

    private static boolean isUnintendedGuardEnabled = false;

    private final List<String> mWhiteList = new ArrayList<>();
    private boolean mSealed = false;

    public AccessibilityConfig() {
        if (isUnintendedGuardEnabled()) {
            mWhiteList.add(DeveloperUtils.selfPackage());
        }
    }

    public static boolean isUnintendedGuardEnabled() {
        return isUnintendedGuardEnabled;
    }

    public static void setIsUnintendedGuardEnabled(boolean isUnintendedGuardEnabled) {
        AccessibilityConfig.isUnintendedGuardEnabled = isUnintendedGuardEnabled;
    }

    public boolean whiteListContains(String packageName) {
        return mWhiteList.contains(packageName);
    }

    public void addWhiteList(String packageName) {
        if (mSealed)
            throw new IllegalStateException("Sealed");
        mWhiteList.add(packageName);
    }

    public final void seal() {
        mSealed = true;
    }
}

File: app/src/main/java/org/autojs/autojs/runtime/accessibility/AccessibilityConfig.kt
package org.autojs.autojs.runtime.accessibility

import org.autojs.autojs.core.pref.Pref
import org.autojs.autojs.util.DeveloperUtils

/**
 * Created by Stardust on Apr 29, 2017.
 * Modified by SuperMonster003 as of Apr 2, 2024.
 * Transformed by SuperMonster003 on Apr 2, 2024.
 */
class AccessibilityConfig {

    private val mBlacklist = ArrayList<String>()
    private var mSealed = false

    init {
        if (isUnintendedGuardEnabled()) {
            addBlacklist(DeveloperUtils.selfPackage())
        }
    }

    fun isInBlacklist(packageName: String) = mBlacklist.contains(packageName)

    fun addBlacklist(packageName: String) {
        check(!mSealed) { "Sealed" }
        mBlacklist.add(packageName)
    }

    fun seal() {
        mSealed = true
    }

    companion object {

        @JvmStatic
        fun isUnintendedGuardEnabled() = Pref.isGuardModeEnabled

    }

}
